internal/metrics/prometheus.go

```diff
@@ -176,6 +238,11 @@ func (c *CtrlFuncMetrics) RecordTranslationFailure() {
 	}).Inc()
 }
 
+// RecordTranslationBrokenResources records the number of resources failing translation.
+func (c *CtrlFuncMetrics) RecordTranslationBrokenResources(count float64) {
```

Comment:
The only reason I understand we need the `float64` is because gauge uses it as it's underlying type. Since you can't really have non integer amount of broken resources I'd suggest to use `int` (or `uint`) throughout the call stack and only cast to `float64` when interacting with the gauge (e.g. in `.Set`)